lit_nlp/client/modules/perturbations_table_module.ts

```diff
@@ -0,0 +1,310 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// tslint:disable:no-new-decorators
+import '@material/mwc-icon';
+
+import * as d3 from 'd3';
+import {customElement, html} from 'lit-element';
+import {computed, observable} from 'mobx';
+import {classMap} from 'lit-html/directives/class-map';
+import {styleMap} from 'lit-html/directives/style-map';
+
+import '../elements/text_diff';
+import {app} from '../core/lit_app';
+import {LitModule} from '../core/lit_module';
+import {TableData} from '../elements/table';
+import {CallConfig, FacetMap, GroupedExamples, IndexedInput, LitName, ModelsMap, Spec} from '../lib/types';
+import {doesOutputSpecContain, formatLabelNumber, findSpecKeys} from '../lib/utils';
+import {GroupService} from '../services/group_service';
+import {DeltasService, RegressionService, ClassificationService, SliceService} from '../services/services';
+import {RegressionInfo} from '../services/regression_service';
+import {DeltaRow, DeltaInfo, Source} from '../services/deltas_service';
+
+import {styles} from './perturbations_table_module.css';
+import {styles as sharedStyles} from './shared_styles.css';
+
+type DeltaRowsById = {
+  [id: string]: DeltaRow
+};
+
+/**
+ * Module to sort generated countefactuals by the change in prediction for a
+ regression or multiclass classification model.
+ */
+@customElement('perturbations-table-module')
+export class PerturbationsTableModule extends LitModule {
+  static title = 'Perturbations';
+  static numCols = 4;
+  static duplicateForModelComparison = true;
+  static duplicateAsRow = false;
+  static template = (model = '') => {
+    return html`<perturbations-table-module model=${model}></perturbations-table-module>`;
+  };
+
+  static shouldDisplayModule(modelSpecs: ModelsMap, datasetSpec: Spec) {
```

Comment:
we usually place this method at bottom of module class

Reply:
done